docs/dorgeclipseopenj9criuimmutableenvvars.md

```diff
@@ -0,0 +1,57 @@
+<!--
+* Copyright (c) 2017, 2023 IBM Corp. and others
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution and is available at
+* https://www.eclipse.org/legal/epl-2.0/ or the Apache
+* License, Version 2.0 which accompanies this distribution and
+* is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* This Source Code may also be made available under the
+* following Secondary Licenses when the conditions for such
+* availability set forth in the Eclipse Public License, v. 2.0
+* are satisfied: GNU General Public License, version 2 with
+* the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] https://openjdk.org/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# -Dorg.eclipse.openj9.criu.ImmutableEnvVars
+
+This system property specifies the environment variables that the VM can detect on restore.
+
+:fontawesome-solid-triangle-exclamation:{: .warn aria-hidden="true"} **Restrictions:** This option takes effect only when the `-XX:+EnableCRIUSupport` option is enabled. This option can be used only on restore.
+
+## Syntax
+
+        -Dorg.eclipse.openj9.criu.ImmutableEnvVars=<environment_variables>
+
+Where `<environment_variables>` is a comma-separated list of environment variables.
+
+The following environment variables are included in the list by default and cannot be removed:
+
+- `LANG`
+- `LC_ALL`
+- `LC_CTYPE`
+
+## Explanation
+
+The environment variables are a property of the operating system environment and not the VM. The VM considers these properties as immutable and does not allow the environment variables to change values between checkpoint and restore. Even though the environment variables are always immutable, but with the CRIU support, you can take a checkpoint in one environment and restore in another.
+
+At the checkpoint time, the VM can detect the environment variables that are in the `immutableEnvvars` list. You can add any other immutable variables that you want the VM to detect at restore in the `immutableEnvvars` list with the `-Dorg.eclipse.openj9.criu.ImmutableEnvVars` option. If an environment variable is not in the list, the VM does not detect that variable even if the underlying system has defined them. These environment variables exist for the entire lifetime of the VM (checkpoint and restore).
+
```

Comment:
Should we use an example here (perhaps the same could be asked for different options) to make it clear what the syntax is when one wants to specify a new immutable env var ?

Reply:
Could you please provide real example for the same? We would like to avoid values like [ENV_VAR1,ENV_VAR2,...ENV_VARN]. Thanks!

Reply:
May be better answered by @tajila to ensure that I don't get the precise syntax wrong.

Reply:
`-Dorg.eclipse.openj9.criu.ImmutableEnvVars=INVOKED,WLP_USER_DIR,WLP_OUTPUT_DIR,LOG_DIR,X_LOG_DIR,LOG_FILE,X_LOG_FILE,VARIABLE_SOURCE_DIRS,X_CMD`